src/components/AccountDrawer/UniwalletModal.tsx

```diff
@@ -39,44 +37,35 @@ const Divider = styled.div`
 `
 
 export default function UniwalletModal() {
-  const open = useModalIsOpen(ApplicationModal.UNIWALLET_CONNECT)
-  const toggle = useToggleUniwalletModal()
-
+  const { pendingConnection, error, cancelActivation } = useActivateConnection()
   const [uri, setUri] = useState<string>()
+
+  const isUniwapWallet = pendingConnection?.type === ConnectionType.UNIWALLET
+  // Displays the modal if a Uniswap Wallet Connection is pending without error, & qrcode URI is available
+  const open = isUniwapWallet && error === undefined && uri !== undefined
```

Comment:
A few things:
1. Can we manage the state inside `useActivateConnection` in a way that error is automatically cleared when connection is pending, so that here, we can actually use `isPending` instead of `error === undefined`? Just feels a bit more idiomatic 
2. Can `uri` be null and if yes, does that have any other meaning? If not, maybe a simple `&& !!uri` does the trick.

Reply:
1. Yes, I updated to use a stricter `ActivationState` type that makes this cleaner.

2. Technically the WalletConnect eventListener returns `any` for the uri. It's possible it could return null, would have to look into web3-react and/or the WalletConnect lib but will use `&& !!uri` for now 👍

Reply:
Yeah, in case of 2nd, I think that's a correct assumption, I am not aware of any differentiation there.